### ISSUE TITLE
`tech` - implement immer on all redux states

### ISSUE THREAD

**Issue by thedoublejay** [DeFiCh/wallet#1881]

**Comment by kyleleow** [comment]
Redux toolkit's `createSlice` is already using `Immer` under the hood, so I don't think we need to make any changes, as long as we stick to `createSlice` or `createReducer` in our state mutation logic

Ref: [link](https://redux-toolkit.js.org/usage/immer-reducers#redux-toolkit-and-immer)

**Comment by kyleleow** [comment]
From what i found, redux toolkit's `createSlice()` is using `Immer` internally, so as long as we stick to `createReducer` and `createSlice`, we are safe to `mutate` the state directly by assignment

Ref: 
- [doc](https://redux-toolkit.js.org/usage/immer-reducers#redux-toolkit-and-immer)
- [SO question](https://stackoverflow.com/questions/69575694/when-to-update-state-or-return-action-payload-for-redux-thunk-extra-reducers)

**Comment by thedoublejay** [comment]
@kyleleow Yup, I created this ticket because some states are still using spread operator for the reducers.
